server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -372,25 +381,37 @@ public static Query dateRangeQuery(
             boolean includeLower,
             boolean includeUpper,
             @Nullable ZoneId timeZone,
-            DateMathParser parser,
+            DateMathParser forcedDateParser,
+            DateMathParser fallbackParser,
             QueryShardContext context,
             Resolution resolution,
             BiFunction<Long, Long, Query> builder
         ) {
             return handleNow(context, nowSupplier -> {
                 long l, u;
+                DateMathParser parser = forcedDateParser == null ? fallbackParser : forcedDateParser;
                 if (lowerTerm == null) {
                     l = Long.MIN_VALUE;
                 } else {
-                    l = parseToLong(lowerTerm, !includeLower, timeZone, parser, nowSupplier, resolution);
+                    if (lowerTerm instanceof Number && forcedDateParser == null) {
+                        // force epoch_millis
+                        l = resolution.convert(Instant.ofEpochMilli(((Number) lowerTerm).longValue()));
```

Comment:
I think we need "resolution.convert" here to this gets kind of ugly. I personally prefer the earlier variant where we just have a "toString" detour but keep the rest of the code as-is.

Reply:
Yeah. I think you are right. If this were super performance sensitive code I'd argue more, but building queries typically isn't. I think going back is probably the right thing.